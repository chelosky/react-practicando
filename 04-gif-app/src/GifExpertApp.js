import React, {useState} from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Dragon Ball Z'])

    // const handleAdd = () => {
    //     // setCategories([...categories, 'Hunter x Hunter']);
    //     setCategories( item => [...item, 'Hunter x Hunter']);
    // };
    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories }  />
            <hr />
             
            <ol>
                {
                    categories.map((category) => {
                        return <GifGrid
                                    key={ category }
                                    category={category}
                                />;
                    })
                }
            </ol>
        </>
    );
}

export default GifExpertApp;