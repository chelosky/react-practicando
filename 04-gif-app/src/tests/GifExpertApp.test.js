import React from "react";
import "@testing-library/jest-dom";
import GifExpertApp from '../GifExpertApp';
import { shallow } from "enzyme";

describe('GifExpertApp', ()=> {
    test('should be created correctly', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });
    
    test('should show list default categories', () => {
        const defaultCategories = ['Dragon Ball Z', 'Kaladin'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ defaultCategories } />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(defaultCategories.length);
        
    })
    
})