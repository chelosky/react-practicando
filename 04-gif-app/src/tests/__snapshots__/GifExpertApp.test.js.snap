// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp should be created correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`GifExpertApp should show list default categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball Z"
      key="Dragon Ball Z"
    />
    <GifGrid
      category="Kaladin"
      key="Kaladin"
    />
  </ol>
</Fragment>
`;
